pkgs/by-name/do/doubleratchet/package.nix

```
@@ -0,0 +1,41 @@
+{
+  python3Packages,
+  lib,
+  fetchFromGitHub,
+  nix-update-script,
+}:
+python3Packages.buildPythonPackage rec {
+  pname = "doubleratchet";
+  version = "1.1.0";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "Syndace";
+    repo = "python-doubleratchet";
+    tag = "v${version}";
+    hash = "sha256-yoph3u7LjGjSPi1hFlXzWmSNkCXvY/ocTt2MKa+F1fs=";
+  };
+
+  strictDeps = true;
+
+  passthru.updateScript = nix-update-script { };
+
+  nativeBuildInputs = with python3Packages; [ setuptools ];
+
+  propagatedBuildInputs = with python3Packages; [
+    cryptography
+    pydantic
+    typing-extensions
+  ];
```

See [contributing guidelines for python](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#contributing-guidelines-contributing-guidelines) and [handling dependencies](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#handling-dependencies-handling-dependencies):

> The two parameters, `pyproject` and `build-system` are set to avoid the legacy setuptools/distutils build.

```suggestion
  build-system = [
    setuptools
  ];

  dependencies = [
    cryptography
    pydantic
    typing-extensions
  ];
```